src/middlewared/middlewared/plugins/jail.py

```diff
@@ -383,7 +384,7 @@ def get_activated_pool(self):
         """Returns the activated pool if there is one, or None"""
         try:
             pool = ioc.IOCage(skip_jails=True).get('', pool=True)
-        except RuntimeError as e:
+        except (RuntimeError, SystemExit) as e:
```

Comment:
In this case iocage will raise a different exception now.  It raises RuntimeError (SystemExit if in foreground) if there are more than one, otherwise a PoolNotActivatedError

Reply:
@skarekrow i don't think `PoolNotActivated` is raised because of this
```
if force_raise:
     raise callback_exception(message)
else:
     raise SystemExit(1)
```
And `force_raise` is always `False`,
```
ocage_lib.ioc_common.logit(
                                {
                                    'level': 'EXCEPTION',
                                    'message': 'Please specify a pool to'
                                    ' activate with iocage activate POOL'
                                },
                                _callback=self.callback,
                                silent=self.silent,
                                exception=ioc_exceptions.PoolNotActivated)
```

Reply:
We should actually use force raise in the middleware. Will allow us to stop parsing for SystemExit and catch the actual exceptions iocage is raising. Gives us more insight ;)